tar/bsdtar.c

```diff
@@ -1315,62 +1319,65 @@ configfile_helper(struct bsdtar *bsdtar, const char *line)
 	 * duplication, but to reduce the number of diffs to a later version,
 	 * we'll do it here.
 	 */
-	len = strlen(conf_opt);
+	len = strlen(bsdtar->conf_opt);
 	if ((len > 0) &&
-	    ((conf_opt[len - 1] == ' ') || (conf_opt[len - 1] == '\t'))) {
+	    ((bsdtar->conf_opt[len - 1] == ' ') ||
+	     (bsdtar->conf_opt[len - 1] == '\t'))) {
 		bsdtar_warnc(bsdtar, 0,
 		    "option contains trailing whitespace; future behaviour"
 		    " may change for:\n    %s", line);
 	}
 
 	/* Split line into option and argument if possible. */
-	optlen = strcspn(conf_opt, " \t");
+	optlen = strcspn(bsdtar->conf_opt, " \t");
 
 	/* Is there an argument? */
-	if (conf_opt[optlen]) {
+	if (bsdtar->conf_opt[optlen]) {
 		/* NUL-terminate the option name. */
-		conf_opt[optlen] = '\0';
+		bsdtar->conf_opt[optlen] = '\0';
 
 		/* Find the start of the argument. */
-		conf_arg = conf_opt + optlen + 1;
-		conf_arg += strspn(conf_arg, " \t");
+		bsdtar->conf_arg = bsdtar->conf_opt + optlen + 1;
```

Comment:
You're setting `bsdtar->conf_arg` to a string which was not allocated by malloc.  What happens if we exit from this point?